beacon-chain/blockchain/service.go

```diff
@@ -237,19 +229,6 @@ func (s *Service) StartFromSavedState(saved state.BeaconState) error {
 			return errors.Wrap(err, "could not set finalized block as validated")
 		}
 	}
-	s.headLock.RLock()
```

Comment:
It is tangentially, the problem is that fillInForkChoice needs to change the API, but since this call is irrelevant since head is anyway finalized, I decided to clean this up in this same call